### ISSUE TITLE
CouchbaseSample.iOS goes into bad state after "Cleaning"

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-net#556]
Using CBL.NET (d422a5ccc89407edfd196c7fbdf86629f5a32ce5)

To repro,
1. Create and replicate a few docs
2. Select a few items and "Clean"
3. "Cleaned" items remain and clicking on them creates error messages. 

TODO
- [ ] Need to verify if an app / ui issue

**Comment by borrrden** [comment]
I'm not able to reproduce using those steps.  The items disappear when I press the clean button.

**Comment by sethrosetter** [comment]
To repro:
1. Deploy 2 clients, one Android, one iOS, targeting a local sync_gateway
2. Add a few docs from each client, make sure they sync
3. Check 3-4 (not sure if muliple are required) docs on the Android client
4. Select "Clean" on the iOS client.

Checked items are removed on the Android client but remain in the iOS TableView. Clicking on the selected items on the iOS side will cause the app to err. 

This may be an issue with Android push replication.

**Comment by borrrden** [comment]
In the end it *was* an error with Android pushing, but not specifically Android pushing.  The thing is, we were using ForestDB on Android and it had a bug in the way it created a local revision history.  

LONG EXPLANATION:
When a \_revs\_diff response is received from Sync Gateway, it sometimes comes with information about possible ancestors.  The client then takes these local ancestors and runs through the parent revisions of the revision to be pushed until it finds a matching revision and uses that as a revision history that is then sent back to Sync Gateway via \_bulk\_docs.  The problem was that when it finds the matching revision ID it should also add it into the history.  The client was not adding the final entry, thus causing it to appear as a conflict instead.  In my simple repro example, I was using rev1 and rev2, and both would appear as root revisions because rev1 was actually a root revision, but rev2 didn't include the final entry (rev1) in its revision history.  This conflict propagated down to the iOS client and got inserted correctly as a conflict.  However, because it is a conflict when it was deleted iOS reverted back to the other leaf (losing conflict) revision and displayed that instead.  This caused havoc when trying to update it further because it was not listed as a current revision and so the client thought it was trying to create a new revision on one that didn't exist.

TL;DR
Forgot to include the final revision in the revision history for a local ForestDB database and ended up causing conflicts that didn't actually exist in sync gateway.

Fix coming today.
